addons/account/models/account_move.py

```diff
@@ -683,8 +683,17 @@ def _compute_name(self):
                 # it. We only check the first move as an approximation (enough for new in form view)
                 move._set_next_sequence()
             elif move.quick_edit_mode and not move.posted_before:
-                # We always suggest the next sequence as the default name of the new move
-                move._set_next_sequence()
+                if not move.name:
+                    move._set_next_sequence()
+                else:
+                    _, seq_format_values = self._get_sequence_format_param(move.name)
+                    seq_year = seq_format_values['year']
+                    seq_month = seq_format_values['month']
+
+                    year = move.date.year % (10 ** seq_format_values['year_length'])
+                    if seq_year and seq_year != year or seq_month and seq_month != move.date.month:
```

Comment:
This seems to be the same as doing `_constrains_date_sequence` like below, I think that combining the to `elif` cases would lead to cleaner code

Reply:
This would indeed be much cleaner, however `_constrains_date_sequence` does not apply in quick edit mode:
https://github.com/odoo/odoo/blob/4c19af2b4b1af25a7ee0d093e1bc9ae3c8853b69/addons/account/models/account_move.py#L2401-L2403
https://github.com/odoo/odoo/blob/179f4f83db0d029850375012bbf80dedc6679505/addons/account/models/sequence_mixin.py#L54-L63

Reply:
Maybe try to refactor a bit then? Something like this? (didn't try)
```diff
diff --git a/addons/account/models/sequence_mixin.py b/addons/account/models/sequence_mixin.py
index 5bd725d241b..68b89edc71f 100644
--- a/addons/account/models/sequence_mixin.py
+++ b/addons/account/models/sequence_mixin.py
@@ -50,6 +50,15 @@ class SequenceMixin(models.AbstractModel):
     def _must_check_constrains_date_sequence(self):
         return True
 
+    def _sequence_matches_date(self):
+        date = fields.Date.to_date(self[self._sequence_date_field])
+        sequence = self[self._sequence_field]
+        format_values = self._get_sequence_format_param(sequence)[1]
+        return sequence and date and not (
+            format_values['year'] and format_values['year'] != date.year % 10**len(str(format_values['year']))
+            or format_values['month'] and format_values['month'] != date.month
+        )
+
     @api.constrains(lambda self: (self._sequence_field, self._sequence_date_field))
     def _constrains_date_sequence(self):
         # Make it possible to bypass the constraint to allow edition of already messed up documents.
@@ -64,11 +73,7 @@ class SequenceMixin(models.AbstractModel):
             date = fields.Date.to_date(record[record._sequence_date_field])
             sequence = record[record._sequence_field]
             if sequence and date and date > constraint_date:
-                format_values = record._get_sequence_format_param(sequence)[1]
-                if (
-                    format_values['year'] and format_values['year'] != date.year % 10**len(str(format_values['year']))
-                    or format_values['month'] and format_values['month'] != date.month
-                ):
+                if not record._sequence_matches_date():
                     raise ValidationError(_(
                         "The %(date_field)s (%(date)s) doesn't match the sequence number of the related %(model)s (%(sequence)s)\n"
                         "You will need to clear the %(model)s's %(sequence_field)s to proceed.\n"
```

Reply:
I actually did something similar, and I thought I had pushed the commit before pinging you, but it seems I didn't 😅.